backend/dal.keycloak/Partials/PimsKeycloakUserService.cs

```diff
@@ -173,6 +173,7 @@ public async Task<Entity.User> UpdateUserAsync(Entity.User user)
 
             // Now update keycloak
             var kmodel = _mapper.Map<KModel.UserModel>(user);
+            kmodel.Id = user.KeycloakUserId.Value;
```

Comment:
I shouldn't have to do this, but the mapping doesn't appear to be correctly handling this.